test/fiber/scheduler.rb

```diff
@@ -127,11 +127,19 @@ def kernel_sleep(duration = nil)
   end
 
   # Used when blocking on synchronization (Mutex#lock, Queue#pop, SizedQueue#push, ...)
-  def block(blocker)
-    # p [__method__, blocker]
+  def block(blocker, timeout = nil)
+    # p [__method__, blocker, timeout]
     @locking += 1
+    if timeout
+      @waiting[Fiber.current] = current_time + timeout
```

Comment:
Do we still need to change `@locking` if there is a `timeout`?

Reply:
Implementation detail.

Reply:
Doesn't answer my question :p
I think we don't right?
If so then we can probably simplify the code to have a single `if timeout` at the top of the method.
The test scheduler is also an example so I think it's important to be clear and show how to implement.

Reply:
What I was saying is there are multiple valid ways to implement it and I don't see any significant advantage of one way over another. If you have a strong opinion about it, we can revise it. However, this was previously `mutex_lock` and then `@locking` was supposed to be how many things are waiting on locks. We should rename it to `blocking` or `blocked` or something too.

Reply:
Renamed to `@blocking`.